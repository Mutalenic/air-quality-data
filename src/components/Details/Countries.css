.reloadText {
  font-size: x-large;
}

.searchCountry {
  border: 1px solid grey;
  border-radius: 5px;
  height: 20px;
  width: 50%;
  padding: 2px 23px 2px 30px;
  outline: 0;
  background-color: #f5f5f5;
}

.searchCountry:hover,
.search:focus {
  border: 1.5px solid #009688;
  background-color: white;
}
