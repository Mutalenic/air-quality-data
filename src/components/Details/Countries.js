import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faRefresh } from '@fortawesome/free-solid-svg-icons';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Africa from '../Maps/Africa.png';
import Americas from '../../Design/America.png';
import Antarctic from '../Maps/Antarctic.png';
import Asia from '../Maps/Asia.png';
import Europe from '../Maps/Europe.png';
import Oceania from '../Maps/Oceania.png';

const Countries = () => {
  const countries = useSelector((state) => state.countryReducer);
  if (!countries.length) {
    return (
      <div className="m-2">
        <p>Loading</p>
        <NavLink to="/" className="link text-light">
          <p>Click to reload</p>
          <FontAwesomeIcon icon={faRefresh} className="icon" text="reload" />
        </NavLink>
      </div>
    );
  }
  let region1 = countries[0].region;
  if (region1 === 'Africa') region1 = Africa;
  else if (region1 === 'Asia') region1 = Asia;
  else if (region1 === 'Europe') region1 = Europe;
  else if (region1 === 'Oceania') region1 = Oceania;
  else if (region1 === 'Americas') region1 = Americas;
  else region1 = Antarctic;
  return (
    <div>
      <Header id="/" />
      <div className="row m-4">
        <div className="col-12 d-flex justify-content-center align-items-center border gap-4">
          <h3>{countries[0].region}</h3>
          <img src={region1} alt="" className="m-2 img1" />
        </div>
        {
          countries.map((country) => (
            <Country
              key={country.code}
              id={country.code}
              name={country.name.common}
              lat={country.latlng[0]}
              lng={country.latlng[1]}
              population={country.population}
              region={country.region}
              flag={country.flag}
            />
          ))
        }
      </div>
    </div>
  );
};

export default Countries;
