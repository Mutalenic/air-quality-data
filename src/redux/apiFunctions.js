export const fetchCountries = async (reg) => {
    const res = await fetch('https://restcountries.com/v3.1/all')
      .then((data) => data.json());
    const result = res.filter((country) => country.region === reg);
    return result;
  };
  
  const url = 'https://api.openweathermap.org/data/2.5/air_pollution?';
  const id = '6574f405463f1e3a64b32c567ddd4bc8';
  
  export const getPollutionInfor = async (lat, lon) => {
    const res = await fetch(`${url}lat=${lat}&lon=${lon}&appid=${id}`);
    return res.json();
  };
  