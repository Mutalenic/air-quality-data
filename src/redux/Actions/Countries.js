import { fetchCountries } from '../apiFunctions';

export const ADD_COUNTRIES = 'air-quality-data/Countries/ADD_COUNTRIES';

export const addCountries = (payload) => ({
  type: ADD_COUNTRIES,
  payload,
});

export const getCountries = (reg) => async (dispatch) => {
  const countries = await fetchCountries(reg);
  dispatch({
    type: ADD_COUNTRIES,
    payload: countries.map((country) => ({
      name: country.name,
      region: reg,
      latlng: country.latlng,
      population: country.population,
      code: country.cca2,
      flag: country.flags.png,
    })),
  });
};
