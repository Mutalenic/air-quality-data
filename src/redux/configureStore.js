import { legacy_createStore as createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { airPollutionReducer } from './Town/town';
import countriesReducer from './Country/countries';

const rootReducer = combineReducers({
  countriesReducer,
  airPollutionReducer,
});
const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk, logger)));

export default store;
