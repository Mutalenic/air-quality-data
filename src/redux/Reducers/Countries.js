import { ADD_COUNTRIES } from '../Actions/Countries';

const countriesReducer = (state = [], action) => {
  switch (action.type) {
    case ADD_COUNTRIES:
      return action.payload;
    default:
      return state;
  }
};

export default countriesReducer;
