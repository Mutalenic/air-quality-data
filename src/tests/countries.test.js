import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { Provider } from 'react-redux';
import pollutionReducer from '../redux/Reducers/Pollution';
import countriesReducer from '../redux/Reducers/Countries';
import Countries from '../components/Details/Countries';

const rootReducer = combineReducers({
  countriesReducer,
  pollutionReducer,

});
const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk, logger)));

afterEach(cleanup);

describe('countries list ', () => {
  test('Countries renders correctly', () => {
    const countries = render(
      <Provider store={store}>
        <Router>
          <Countries />
        </Router>
      </Provider>,
    );
    expect(countries).toMatchSnapshot();
  });
});
