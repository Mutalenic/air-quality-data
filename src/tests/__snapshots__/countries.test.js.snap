// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countries list  Countries renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-2"
      >
        <p>
          Loading please wait!
        </p>
        <a
          aria-current="page"
          class="reloadText active"
          href="/"
        >
          <p>
            Click to reload
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrows-rotate icon"
            data-icon="arrows-rotate"
            data-prefix="fas"
            focusable="false"
            role="img"
            text="reload"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 16c-17.67 0-32 14.31-32 32v74.09C392.1 66.52 327.4 32 256 32C161.5 32 78.59 92.34 49.58 182.2c-5.438 16.81 3.797 34.88 20.61 40.28c16.89 5.5 34.88-3.812 40.3-20.59C130.9 138.5 189.4 96 256 96c50.5 0 96.26 24.55 124.4 64H336c-17.67 0-32 14.31-32 32s14.33 32 32 32h128c17.67 0 32-14.31 32-32V48C496 30.31 481.7 16 464 16zM441.8 289.6c-16.92-5.438-34.88 3.812-40.3 20.59C381.1 373.5 322.6 416 256 416c-50.5 0-96.25-24.55-124.4-64H176c17.67 0 32-14.31 32-32s-14.33-32-32-32h-128c-17.67 0-32 14.31-32 32v144c0 17.69 14.33 32 32 32s32-14.31 32-32v-74.09C119.9 445.5 184.6 480 255.1 480c94.45 0 177.4-60.34 206.4-150.2C467.9 313 458.6 294.1 441.8 289.6z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-2"
    >
      <p>
        Loading please wait!
      </p>
      <a
        aria-current="page"
        class="reloadText active"
        href="/"
      >
        <p>
          Click to reload
        </p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrows-rotate icon"
          data-icon="arrows-rotate"
          data-prefix="fas"
          focusable="false"
          role="img"
          text="reload"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 16c-17.67 0-32 14.31-32 32v74.09C392.1 66.52 327.4 32 256 32C161.5 32 78.59 92.34 49.58 182.2c-5.438 16.81 3.797 34.88 20.61 40.28c16.89 5.5 34.88-3.812 40.3-20.59C130.9 138.5 189.4 96 256 96c50.5 0 96.26 24.55 124.4 64H336c-17.67 0-32 14.31-32 32s14.33 32 32 32h128c17.67 0 32-14.31 32-32V48C496 30.31 481.7 16 464 16zM441.8 289.6c-16.92-5.438-34.88 3.812-40.3 20.59C381.1 373.5 322.6 416 256 416c-50.5 0-96.25-24.55-124.4-64H176c17.67 0 32-14.31 32-32s-14.33-32-32-32h-128c-17.67 0-32 14.31-32 32v144c0 17.69 14.33 32 32 32s32-14.31 32-32v-74.09C119.9 445.5 184.6 480 255.1 480c94.45 0 177.4-60.34 206.4-150.2C467.9 313 458.6 294.1 441.8 289.6z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
